src/Illuminate/Cache/Repository.php

```diff
@@ -194,55 +194,63 @@ public function pull($key, $default = null)
      * @param  string  $key
      * @param  mixed   $value
      * @param  \DateTimeInterface|\DateInterval|float|int|null  $minutes
-     * @return void
+     * @return bool
      */
     public function put($key, $value, $minutes = null)
     {
         if (is_array($key)) {
-            $this->putMany($key, $value);
+            $result = $this->putMany($key, $value);
 
-            return;
+            return $result;
         }
 
         if (! is_null($minutes = $this->getMinutes($minutes))) {
-            $this->store->put($this->itemKey($key), $value, $minutes);
+            $result = $this->store->put($this->itemKey($key), $value, $minutes);
 
             $this->event(new KeyWritten($key, $value, $minutes));
```

Comment:
Technically we now have the result variable to tell us if it was written successfully so this event shouldn't be fired if `$result === false`.